### PR TITLE
Move to Cosmos SDK v0.38

### PR DESCRIPTION
**Pull request by gsora**
This PR is the natural continuation of #83: pointing our stack to the latest Cosmos SDK available.

Cosmos SDK v0.38 contains a [substantial amount of changes](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.38.0), porting our code to this version has been an emotional roller-coaster.

Lots of new under-the-hood changes (mainly security and technical debt-related), but nothing that inherently breaks the current Commercio genesis file or state machine _as far as I see_.

The two new functional changes I've seen are:

 - errors are now complying with Go 1.13 wrap/unwrap interface, hence the SDK now defines and handles errors in a more idiomatic way
 - the keyring has been completely replaced by an implementation based on the [99designs/keyring](https://github.com/99designs/keyring) package, which leverages native operating system-level keyring facilities like macOS Keychain or KDE KWallet.

---

## Checklist
- [x] Targeted PR against correct branch
- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For Admin Use:
- [ ] Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- [ ] Reviewers Assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]"

### CODE REVIEW

**Comment by gsora** [comment]
**PSA**: moving to Cosmos SDK v0.38 keyring is __not__ an [automated process](https://github.com/cosmos/cosmos-sdk/pull/5097)!

Users __must__ be advised to migrate their existing keys using the `cncli keys migrate` command.

Ledger users should not be affected by this change.

**Comment by gsora** [comment]
Since Tendermint v0.33 - which Cosmos SDK v0.38 uses - some `evidence` parameters have been changed.

Before executing any migration, users **must** be advised to run the following shell oneliner to make the genesis compatible with Tendermint v0.33:

```bash
$ jq 'setpath(["consensus_params", "evidence"]; {"max_age_num_blocks":"100000","max_age_duration":"172800000000000"})' [genesis-file-path] > new_genesis.json
```

Standard `cnd` migrations can be then used successfully.

File: cmd/cnd/main.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"encoding/json"
 	"io"
 
+	"github.com/cosmos/cosmos-sdk/client/flags"
```
Comment (alessio) [inline cmd/cnd/main.go]: `goimports` please :)

File: x/ante/ante_test.go [hunk 2]
```diff
@@ -23,14 +25,13 @@ func checkValidTx(t *testing.T, anteHandler sdk.AnteHandler, ctx sdk.Context, tx
...
+func checkInvalidTx(t *testing.T, anteHandler sdk.AnteHandler, ctx sdk.Context, tx sdk.Tx, simulate bool, code error) {
 	_, err := anteHandler(ctx, tx, simulate)
 	require.NotNil(t, err)
 
-	result := sdk.ResultFromError(err)
-
-	require.Equal(t, code, result.Code, fmt.Sprintf("Expected %v, got %v", code, result))
-	require.Equal(t, sdk.CodespaceRoot, result.Codespace)
+	require.True(t, errors.Is(sdkErr.ErrInsufficientFee, code))
+	//require.Equal(t, code, result.Code, fmt.Sprintf("Expected %v, got %v", code, result))
```
Comment (alessio) [inline x/ante/ante_test.go]:
```suggestion
```

File: x/ante/ante_test.go [hunk 3]
```diff
@@ -23,14 +25,13 @@ func checkValidTx(t *testing.T, anteHandler sdk.AnteHandler, ctx sdk.Context, tx
...
+	//require.Equal(t, sdk.CodespaceRoot, result.Codespace)
```
Comment (alessio) [inline x/ante/ante_test.go]:
```suggestion
```

File: x/commerciomint/internal/keeper/keeper_test.go [hunk 4]
```diff
@@ -143,7 +145,13 @@ func TestKeeper_OpenCdp(t *testing.T) {
...
 
 			err := k.OpenCdp(ctx, test.owner, test.amount)
-			require.Equal(t, test.error, err, "Invalid error: %s, %s", test.error, err)
+			if test.error != nil {
+				require.Equal(t, test.error.Error(), err.Error())
+			} else {
+				require.NoError(t, err)
+			}
+
+			//require.Equal(t, test.error, err, "Invalid error: %s, %s", test.error, err)
```
Comment (alessio) [inline x/commerciomint/internal/keeper/keeper_test.go]:
```suggestion
```

File: x/commerciomint/client/cli/tx.go [hunk 5]
```diff
@@ -80,14 +84,15 @@ func closeCDPCmd(cdc *codec.Codec) *cobra.Command {
...
+	cmd = flags.PostCommands(cmd)[0]
 
 	return cmd
 }
 
 func closeCDPCmdFunc(cmd *cobra.Command, args []string, cdc *codec.Codec) error {
-	cliCtx := context.NewCLIContext().WithCodec(cdc)
-	txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+	inBuf := bufio.NewReader(cmd.InOrStdin())
+	cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
```
Comment (alessio) [inline x/commerciomint/client/cli/tx.go]: :+1:

File: x/memberships/client/cli/tx.go [hunk 6]
```diff
@@ -39,7 +43,8 @@ func getCmdVerifyUser(cdc *codec.Codec) *cobra.Command {
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
```
Comment (alessio) [inline x/memberships/client/cli/tx.go]:
```suggestion
			inBuf := bufio.NewReader(cmd.InOrStdin())
			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
```

File: x/memberships/client/cli/tx.go [hunk 7]
```diff
@@ -39,7 +43,8 @@ func getCmdVerifyUser(cdc *codec.Codec) *cobra.Command {
...
-			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			inBuf := bufio.NewReader(cmd.InOrStdin())
```
Comment (alessio) [inline x/memberships/client/cli/tx.go]:
```suggestion
```

**Pull request merged**